### PR TITLE
Support creating directories on erasure coded backend

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Support creating directories on erasure coded backend

## Motivation and Context
This PR continues from #5049 where we started supporting
directories for erasure coded backend

## How Has This Been Tested?
Manually using go test and rexray/s3fs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by harshavardhana** [comment]
@balamurugana @poornas can i get some reviews?

File: cmd/xl-v1-object.go [hunk 1]
```diff
@@ -33,6 +33,29 @@ import (
 // list all errors which can be ignored in object operations.
 var objectOpIgnoredErrs = append(baseIgnoredErrs, errDiskAccessDenied)
 
+// putObjectDir hints the bottom layer to create a new directory.
+func (xl xlObjects) putObjectDir(bucket, object string, writeQuorum int) error {
+	var wg = &sync.WaitGroup{}
+
+	errs := make([]error, len(xl.storageDisks))
+	// Prepare object creation in a all disks
```
Comment (poornas) [inline cmd/xl-v1-object.go]: in a all disks -> redundant "a'

File: cmd/xl-v1-object.go [hunk 2]
```diff
@@ -354,6 +383,41 @@ func (xl xlObjects) getObject(bucket, object string, startOffset int64, length i
 	return nil
 }
```
Comment (poornas) [inline cmd/xl-v1-object.go:385]: function needs a comment

**Review by poornas — CHANGES_REQUESTED** [review]

**Comment by poornas** [comment]
How is clean up of these empty directories supposed to work? mc rm will not delete these files. 
```curl -X DELETE localhost:9000/tt1/a1/a2/``` deletes the a2  subdirectory for FS whereas in XL it  does not do so.

**Comment by harshavardhana** [comment]
> How is clean up of these empty directories supposed to work? mc rm will not delete these files.
> curl -X DELETE localhost:9000/tt1/a1/a2/ deletes the a2 subdirectory for FS whereas in XL it does not do so.

It is supposed to delete just like FS @poornas

**Comment by michacassola** [comment]
Do you have a rough estimate when you will be able to solve this to enable multi host multi tenant docker swarm volumes?

**Comment by harshavardhana** [comment]
> Do you have a rough estimate when you will be able to solve this to enable multi host multi tenant docker swarm volumes?

Hopefully soon by end of next week @michacassola

**Comment by harshavardhana** [comment]
> How is clean up of these empty directories supposed to work? mc rm will not delete these files.
> curl -X DELETE localhost:9000/tt1/a1/a2/ deletes the a2 subdirectory for FS whereas in XL it does not do s

Updated the PR with proper delete support as well @poornas @balamurugana  PTAL

**Review by poornas — APPROVED** [review]
Tested locally. LGTM.

**Review by balamurugana — APPROVED** [review]
LGTM

**Pull request merged**